sycl/source/detail/persistent_device_code_cache.hpp

```diff
@@ -0,0 +1,161 @@
+//==---------- persistent_device_code_cache.hpp -----------------*- C++-*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/device_binary_image.hpp>
+#include <CL/sycl/detail/pi.hpp>
+#include <CL/sycl/detail/util.hpp>
+#include <CL/sycl/device.hpp>
+#include <chrono>
+#include <fcntl.h>
+#include <string>
+#include <sys/stat.h>
+#include <thread>
+#include <vector>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+/* This is temporary solution until std::filesystem is available when SYCL RT
+ * is moved to c++17 standard*/
+std::string getDirName(const char *Path);
+
+/// Checks if specified path is present
+inline bool isPathPresent(const std::string &Path) {
+  struct stat Stat;
+  return !stat(Path.c_str(), &Stat);
+}
+
+/// Make directory recursibely
+int makeDir(const char *Dir);
+
+class LockCacheItem {
```

Comment:
please add comments and usage model (e.g. a check for 'Owned' is needed before use).

Reply:
done